src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -1090,6 +1088,19 @@ DeletePartitionRow(Oid distributedRelationId)
 }
 
 
+/*
+ * PartitionTupleRelationId returns the logicalrelid column in the given
+ * pg_dist_partition heap tuple.
+ */
+Oid
+PartitionTupleRelationId(HeapTuple heapTuple)
+{
+	Form_pg_dist_partition data = (Form_pg_dist_partition) GETSTRUCT(heapTuple);
```

Comment:
We don't need to use `heap_deform_tuple()` here since `logicalrelid` has been in this catalog table since beginning of time, and deforming the whole tuple because of one field seems wasteful.

Reply:
Maybe add that in a comment. We should be very careful doing this from now on, because we can never be sure whether an ALTER EXTENSION was done pre-PG11 or in PG11.

Reply:
Removed this because it wasn't very related to this patch. I'll do it as part of another refactoring PR.